src/app/content/practiceQuestions/components/Answer.tsx

```diff
@@ -81,28 +99,92 @@ export const AnswerBlock = styled.div`
 `;
 
 // tslint:disable-next-line: variable-name
-const Answer = (props: AnswerProps) => {
-  const answerStyle = props.isSelected ? contants.selectedAnswerStyle : contants.unselectedAnswerStyle;
+const StyledAnswerResult = styled.div`
+  ${textRegularStyle}
+  color: ${(props: {variant: PracticeQuestionStyles['label']}) => practiceQuestionStyles[props.variant].passive};
+`;
 
-  return <AnswerBlock style={answerStyle} onClick={props.onSelect}>
+interface AnswerResultProps {
+  showCorrect: boolean;
+  isSubmitted: boolean;
+  isSelected: boolean;
+  isCorrect: boolean;
+}
+
+// tslint:disable-next-line: variable-name
+const AnswerResult = ({ showCorrect, isSelected, isSubmitted, isCorrect }: AnswerResultProps) => {
+  const resultMsgKey = isCorrect
+    ? 'i18n:practice-questions:popup:correct'
+    : 'i18n:practice-questions:popup:incorrect';
+  let variant: PracticeQuestionStyles['label'] | null = null;
+
+  if (isSubmitted && isSelected) {
+    variant = isCorrect ? 'correct' : 'incorrect';
+  } else if (isSubmitted && showCorrect && isCorrect) {
+    variant = 'correct';
+  }
+
+  return variant
+    ? <StyledAnswerResult variant={variant}>
```

Comment:
variant can be passed as `isCorrect ? 'correct' : 'incorrect'` without the pre-set variable right?

then the previous logic could be combined into a short circuit that returns `null`, remove the ternary in the return value and remove the `let variant`